qos/solana/response_getepochinfo.go

```diff
@@ -57,10 +58,15 @@ type responseToGetEpochInfo struct {
 	epochInfo epochInfo
 }
 
-func (r responseToGetEpochInfo) GetObservation() (observation, bool) {
-	return epochInfoResponseObservation{
-		epochInfo: r.epochInfo,
-	}, true
+// GetObservation returns a Solana Endpoint observation based on an endpoint's response to a `getEpochInfo` request.
+// This method implements the response interface used by the requestContext struct. 
+func (r responseToGetEpochInfo) GetObservation() qosobservations.SolanaEndpointDetails {
```

Comment:
How are you deciding when to using pointer vs value receivers? I'm seeing both being used without a consistent pattern.

Reply:
I try to use value receivers whenever possible, reserving pointer receivers for mutating methods (or e.g. if there is a Mutex in the struct). Will review for inconsistencies.